### DISCUSSION TITLE
Loading "self-contained components" into native host

### DISCUSSION THREAD

**Discussion by RichardYu1** [dotnet/runtime#68448] [Q&A]
Was it supported currently to load "self-contained components" into native host in Net Core 5.0?

From below discussion thread, it seems it was supported in Net Core 3.0 then in 2020. So was it now supported now?
https://github.com/dotnet/runtime/issues/35329

We need to use native host because we would like to add /DEPENDENTLOADFLAG linker flag into main executable in order to mitigate DLL sideloading issue so that the app is more secure.

**Comment by vitek-karas** [comment]
If you want to run an app, just need to customize the native code which runs before it starts the runtime/managed code, then you can do self-contained just fine. Just use `hosftx_initialize_for_dotnet_command_line` and then you can either execute the app or get a delegate (depends on what you want).
Alternatively you can basically copy the apphost we build https://github.com/dotnet/runtime/tree/main/src/native/corehost/apphost and customize it - it runs the app (it uses different hostfxr entrypoint, so you can use that one if you want as well as it's just a single call, which ever way you want it).

Issue https://github.com/dotnet/runtime/issues/35465 and related PRs and such added support for the combination of `hostfxr_initialize_for_dotnet_command_line` and getting a delegate - so that you can avoid calling `Main` and instead call some different method you choose.

Note that SDK doesn't have a supported way of building self-contained components, it can only build self-contained apps.
